Source/Private/ASDisplayNode+UIViewBridge.mm

```diff
@@ -796,15 +796,44 @@ - (void)setBackgroundColor:(UIColor *)newBackgroundColor
 - (UIColor *)tintColor
 {
   _bridge_prologue_read;
-  ASDisplayNodeAssert(!_flags.layerBacked, @"Danger: this property is undefined on layer-backed nodes.");
-  return _getFromViewOnly(tintColor);
+  if (_loaded(self)) {
+    if (_flags.layerBacked) {
+        // The first nondefault tint color value in the view’s hierarchy, ascending from and starting with the view itself.
+        return _tintColor ?: self.supernode.tintColor;
+      } else {
+        return _getFromViewOnly(tintColor);
+      }
+  } else {
+    if (_flags.layerBacked) {
+      return _tintColor;
+    } else {
+      return ASDisplayNodeGetPendingState(self).tintColor;
+    }
+  }
 }
 
 - (void)setTintColor:(UIColor *)color
 {
   _bridge_prologue_write;
-  ASDisplayNodeAssert(!_flags.layerBacked, @"Danger: this property is undefined on layer-backed nodes.");
-  _setToViewOnly(tintColor, color);
+  if (_loaded(self)) {
+    if (_flags.layerBacked) {
+      if (![_tintColor isEqual:color]) {
+        _tintColor = color;
+        // Tint color has changed. Unlock here before calling subclasses and exit-early
+        AS::MutexLocker unlock(__instanceLock__);
```

Comment:
This might not be safe because the `MutexLocker` (which basically is `std::lock_guard`) will try to unlock again when it's destroyed and yield an exception. Instead I think it'd be easier to just manage the lock manually (i.e remove `_bridge_prologue_write` and this line, and call `lock`/`unlock` directly on `__instanceLock__`.